pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1436,13 +1437,22 @@ impl<'a> DatadirModification<'a> {
     /// All the modifications in this atomic update are stamped by the specified LSN.
     ///
     pub async fn commit(&mut self, ctx: &RequestContext) -> anyhow::Result<()> {
-        let writer = self.tline.writer().await;
+        let mut writer = self.tline.writer().await;
 
         let pending_nblocks = self.pending_nblocks;
         self.pending_nblocks = 0;
 
         if !self.pending_updates.is_empty() {
-            writer.put_batch(&self.pending_updates, ctx).await?;
+            let mut prev_pending_updates = HashMap::new();
+            std::mem::swap(&mut self.pending_updates, &mut prev_pending_updates);
```

Comment:
```suggestion
            let mut prev_pending_updates = std::mem::take(&mut self.pending_updates);
```

Surprised if clippy did not say anything about this. take is essentially what you are doing, implemented for types which `impl Default`.

Reply:
Thanks: https://github.com/neondatabase/neon/pull/6661/commits/3797e1d98aca59f4e2d90368b7e866b683617c8d